rollup/transaction_ingestion.go

```diff
@@ -0,0 +1,172 @@
+package rollup
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"time"
+
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+)
+
+type TxIngestion struct {
+	loopTicker *time.Ticker
+	db         *sqlx.DB
+	signer     types.Signer
+	key        *ecdsa.PrivateKey
+	txpool     *core.TxPool
+}
+
+// Should this have a safety check on cfg.TxIngestionPollInterval?
```

Comment:
Sometimes the config options are sanitized at the time `cli.Context` is being parsed, other times they are sanitized in the module itself. Right now this PR has no option sanitization, so that would be a good thing to add.

Reply:
Ahhh ok that makes sense